core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -140,6 +140,16 @@
         new Setting<>("node.mode", "network", Function.identity(), Property.NodeScope);
     public static final Setting<Boolean> NODE_INGEST_SETTING =
         Setting.boolSetting("node.ingest", true, Property.NodeScope);
+
+    /**
+     * controls whether the node is allowed to persist things like metadata to disk
+     * Note that this does not control whether the node stores actual indices (see
+     * {@link #NODE_DATA_SETTING}). However, if this is false, {@link #NODE_DATA_SETTING}
+     * and {@link #NODE_MASTER_SETTING} must also be false.
+     *
+     */
+    public static final Setting<Boolean> NODE_LOCAL_STORAGE_SETTING = Setting.boolSetting("node.local_storage", true, Property.NodeScope);
```

Comment:
I don't know how to create the setting so that validation fails if `node.master` or `node.data` is set. The validation logic has only access to the new value of the setting and not the settings object.